typing/typecore.ml

```diff
@@ -4613,20 +4679,21 @@ and type_expect_
         exp_attributes = sexp.pexp_attributes;
         exp_env = env }
   | Pexp_while(scond, sbody) ->
-      let cond_env,wh_cond_region =
-        if is_local_returning_expr scond then env, false
-        else Env.add_region_lock env, true
+      let cond_env,wh_cond_region,mode=
+      (* not checking is_local_returning_expr because you should just use exclave *)
+        Env.add_region_lock env, true, mode_region (Value_mode.local)
       in
       let wh_cond =
-        type_expect cond_env mode_local scond
+        type_expect cond_env mode scond
           (mk_expected ~explanation:While_loop_conditional Predef.type_bool)
       in
-      let body_env,wh_body_region =
-        if is_local_returning_expr sbody then env, false
-        else Env.add_region_lock env, true
+      (* local because unit cross modes  *)
```

Comment:
I think this comment is confusing. I would say that it's local because the value isn't going to escape the region. Probably just remove the comment.